src/cli_common.c

```diff
@@ -352,9 +352,17 @@ void parseRedisUri(const char *uri, const char* tool_name, cliConnInfo *connInfo
     path = strchr(curr, '/');
     if (*curr != '/') {
         host = path ? path - 1 : end;
-        if ((port = strchr(curr, ':'))) {
-            connInfo->hostport = atoi(port + 1);
-            host = port - 1;
+        if (*curr == '[') {
+            curr += 1;
+            if ((port = strchr(curr, ']'))) {
+                connInfo->hostport = atoi(port + 2);
```

Comment:
We need to handle cases where the IPv6 address is not followed by a `:` and a port number as well.

Reply:
resolve